pinot-core/src/main/java/org/apache/pinot/core/util/QueryOptions.java

```diff
@@ -33,6 +33,8 @@
   private final boolean _responseFormatSQL;
   private final boolean _preserveType;
   private final boolean _skipUpsert;
+  private final Boolean _enableSegmentTrim;
+  private final Integer _segmentTrimSize;
```

Comment:
No need to use object
```suggestion
  private final boolean _enableSegmentTrim;
  private final int _minSegmentTrimSize;
```